### DISCUSSION TITLE
Automated "immich_postgres pg_dumpall" to local directories on schedule.

### DISCUSSION THREAD

**Discussion by southsko** [immich-app/immich#7722] [Feature Request]
### The feature

I understand that somehow I can use a container to automate to export the DB, but that is not integral with the already complicated compose.  I understand borg, but no.  Can we just have an export on a cron schedule from within the GUI?

For example I run the following once day (0 0 * * *).   

docker exec -t immich_postgres pg_dumpall -c -U postgres | gzip > /mnt/cache/appdata/immich/db_dump/dump.sql.gz

I already have everything in my immich directory backed up, off and on site.  So this and maybe a restore option on a new install would make one nightmare go away for a lot of people.

I recently migrated and thought that this would have saved me a lot of time after I forgot the docker volume was lost and the db was not a mounted volume.  

For now guys... Please backup your db as well as your photos. 

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by mmomjian** [comment]
It's not possible for the Immich container to run commands on other containers without totally changing the permissions granted to the container. I suggest that the best approach is to run your own backup in crontab or something similar like you suggested.

It's not clear to me that it's possible to run pg_dump from within a postgres connection - I don't think that it is.

**Comment by bo0tzz** [comment]
This isn't something for Immich to handle. Take a look at https://immich.app/docs/administration/backup-and-restore, it includes an example of a docker-compose service that does what you want.
